### PR TITLE
fix: sbtc page

### PR DESCRIPTION
**Pull request by BLuEScioN**
## What type of PR is this? (check all applicable)
- [ ] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Slack context - https://hiropbc.slack.com/archives/C03TU42NL05/p1734370346313639
Added sbtc specific UI changes - https://www.figma.com/design/31q1g27k69z1yXb6r6vrRD/Stacks-Explorer-Working-File?node-id=10031-38309&m=dev
Added token image to each tokenid page
Fixed holder balances not accounting for token decimals
Fixed token list total supply figure not accounting for token decimals

Fixed number abbreviation function
Wrote tests for it

## Issue ticket number and link
https://github.com/hirosystems/explorer/issues/1937 

# Checklist:
- [X] I have performed a self-review of my code.
- [X] I have tested the change on desktop and mobile.
- [X] I have added thorough tests where applicable.
- [X] I've added analytics and error logging where applicable.

### CODE REVIEW

File: src/app/_components/PageWrapper.tsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ export function PageWrapper({
           minHeight={'100vh'}
         >
           <NavBar tokenPrice={tokenPrice} />
-          <Flex direction={'column'} mt={10} mb={8} gap={7}>
+          <Flex direction={'column'} mt={30} mb={8} gap={7}>
```
Comment (BLuEScioN) [inline src/app/_components/PageWrapper.tsx:91]: @andresgalante I removed the mt from PageTitle and added it here as I needed the PageTitle to not have any margin as it was making adding additional UI with the PageTitle difficult

File: src/common/utils/utils.ts [hunk 2]
```diff
@@ -286,25 +292,46 @@ export function getNextPageParam(lastPage?: GenericResponseType<any>) {
   return sum;
 }
 
-export const numberToString = (value: number) => {
```
Comment (BLuEScioN) [inline src/common/utils/utils.ts:289]: This was incorrectly handling numbers whose decimals were greater than 3, and not ideally handling trailing zeroes 
image

File: src/app/tokens/TokenRow/index.tsx [hunk 3]
```diff
@@ -39,7 +69,12 @@ export const TokenRow: FC<{
       </Td>
       <Td isNumeric width={'130px'} padding={'10px 16px 10px 20px'}>
         <Text whiteSpace={'nowrap'} textOverflow={'ellipsis'} overflow={'hidden'} fontSize={'15px'}>
-          {numberToString(ftToken.total_supply ? Number(ftToken.total_supply) : 0)}
+          {ftToken.total_supply
+            ? abbreviateNumber(
+                getFtDecimalAdjustedBalance(ftToken.total_supply, ftToken.decimals || 0),
+                2
+              )
+            : 'N/A'}
```
Comment (BLuEScioN) [inline src/app/tokens/TokenRow/index.tsx:77]: I believe N/A is better than 0 as I don't think you can even have a token with a supply of 0

File: src/app/token/[tokenId]/getTokenInfo.ts [hunk 4]
```diff
@@ -44,14 +45,23 @@ async function getBasicTokenInfo(
       throw new Error('Unable to fetch token info for this request');
     }
 
-    console.log('[debug] cache miss, fetching ', `${apiUrl}/metadata/v1/ft/${tokenId}`);
+    // const contractInfoResponse = await fetch(`${apiUrl}/extended/v1/contract/${tokenId}`);
```
Comment (BLuEScioN) [inline src/app/token/[tokenId]/getTokenInfo.ts:48]: This is the code to get the circulating supply from the holders endpoint. I am commenting it out as discussed so the data doesn't show next to the total supply and potenitally show an inconsistency/inaccuracy

File: src/app/token/[tokenId]/PageClient.tsx [hunk 5]
```diff
@@ -17,29 +21,75 @@ export default function PageClient({
...
+            <Tag
+              color="red.600"
+              bg="red.200"
+              border="1px solid var(--stacks-colors-red-500)"
+              pt={1}
+              py={2.5}
+            >
+              <Flex alignItems={'center'} gap={1}>
+                <Icon as={Warning} h={4} w={4} color="red.600" />
+                <Text fontSize={'md'}>Scam</Text>
```
Comment (andresgalante) [inline src/app/token/[tokenId]/PageClient.tsx:73]: I'd also change the word "scam" to "unverified" what do you think?

File: src/common/utils/utils.ts [hunk 6]
```diff
@@ -286,25 +292,46 @@ export function getNextPageParam(lastPage?: GenericResponseType<any>) {
...
+const maxDecimals = 20; // just some arbitrarily large number to preserve precision
+const formatNumber = (num: number, divisor: number, suffix: string, decimals?: number) => {
+  // If decimals is undefined, don't limit decimal places
+  if (decimals == null) {
+    return `${(num / divisor)
+      .toLocaleString(undefined, { maximumFractionDigits: maxDecimals })
+      .replace(/\.0+$/, '')}${suffix}`;
+  }
+
+  if (num < 1 / Math.pow(10, decimals)) {
```
[the remark this answers is not in the document]
Comment (BLuEScioN) [reply src/common/utils/utils.ts:304]: I agree. Not important since this isn't being used for negative numbers rn and probably not in the future

**Pull request merged**